centinel/primitives/dnslib.py

```diff
@@ -15,7 +15,7 @@ def get_ips(host, nameserver=None, record="A"):
     return lookup_domain(host, nameservers=nameservers, rtype=record)
 
 
-def lookup_domain(domain, nameservers=[], rtype="A", timeout=10):
+def lookup_domain(domain, nameservers=[], rtype="A", timeout=2):
```

Comment:
@rpanah, I seem to be having some issues with github's commenting system, so in case you didn't see my comment in the main thread, I am duplicating it here. Let me know what you think and please respond with data if available.

@rpanah, I have responded to each of your points inline, but I'm going to leave an overall summary here. Basically, we gain a lot of value (speed, robust measurements) from leaving the timeout high and doing concurrent requests. I don't find your arguments compelling enough to reduce that value, though I welcome further critiques.

>We are, to some extent. The main usecase for this is to run on dedicated hardware periodically.

I disagree with this. What is a clean pcap to you? By default, we collect everything, then we can strip off what we don't need. This means that any updates, command and control traffic, and anything else are going to be caught in the pcap, which won't be clean (meaning that other concurrent traffic will be captured by centinel.) We can add code to get rid of this, but it's probably not worth it.

Also, I don't buy that this is necessary. What is the value in having a "clean" pcap? Why would that make analysis any easier? To do a robust analysis, we are going to have to apply higher level information (e.g. matching DNS queries and responses by query ID), so it shouldn't make a difference how much stuff is being requested concurrently.

>Not true. We are doing a sync once every hour. Smaller files are transferred more reliably that way.
    Also not true as I ran into this issue while running the benchmark on test nodes in the field.

To clarify, right now we are doing a single big pcap that is collected for the entire experiment (all primitives contained within the pcap). This capture file is going to be the same size regardless of whether we make the request concurrently. Furthermore, if size is an issue, we can split the pcap up using splitcap or tcpdump.

Regardless, my point was that having a large file is better for the filesystem and this is going to be a problem for us on the backend. We shouldn't be talking about that much data though, so I don't buy that this is a problem that we need to solve. What are the sizes of the pcaps you were having problems with, what were the problems, and why do you think the problems are related to the size of the file?